aten/src/ATen/record_function.cpp

```diff
@@ -428,4 +457,41 @@ void RecordFunction::end() {
   active_ = false;
 }
 
+// RecordFunction pre-sampling
+namespace {
+// Whether to try to create RecordFunction on each call (>0) or
+// use pre-sampling (=0)
+std::atomic<int> global_record_all_functions_ {0};
+}
+
+void setRecordAllFunctions() {
+  ++global_record_all_functions_;
+}
+void unsetRecordAllFunctions() {
+  TORCH_CHECK(--global_record_all_functions_ >= 0);
+}
+
+bool shouldRunRecordFunction(bool& pre_sampled) {
```

Comment:
I want to complain about this mutable reference but I recently found out Google changed their C++ style guide to prefer mutable reference output parameters because they can't be null 😂. It's still pretty obscure...

Reply:
I feel that the documentation explaining what is going on is more appropriately placed here, because there are multiple call sites (but you have only added a big doc at one of the call sites; what if a reader comes from another call site?)